### PR TITLE
Add empty tensor initialization to LazyCompressedParameter 

### PR DESCRIPTION
**Pull request by alexm-nm**

### CODE REVIEW

File: vllm/model_executor/layers/parameters/lazy_compressed.py [hunk 1]
```diff
@@ -56,16 +68,28 @@ def unwrap(e):
             if isinstance(e, LazyCompressedParameter):
                 assert ret_storage_format_cls is None or ret_storage_format_cls == e.storage_format_cls
                 ret_storage_format_cls = e.storage_format_cls
-            return e.uncompressed_data if isinstance(
-                e, LazyCompressedParameter) else e
+
+                if e.is_empty:
+                    e.is_empty = False
+                    e.uncompressed_data = torch.rand(size=e.size(),
```
Comment (mgoin) [inline vllm/model_executor/layers/parameters/lazy_compressed.py:74]: no need to generate random data, please use `torch.empty_like` https://pytorch.org/docs/stable/generated/torch.empty_like.html

File: vllm/model_executor/layers/parameters/lazy_compressed.py [hunk 2]
```diff
@@ -56,16 +68,28 @@ def unwrap(e):
...
         rs = func(*tree_map(unwrap, args), **tree_map(unwrap, kwargs))
 
         def wrap(e):
             if isinstance(e,
                           torch.Tensor) and ret_storage_format_cls is not None:
                 return LazyCompressedParameter(
-                    e, storage_format_cls=ret_storage_format_cls)
+                    e,
+                    is_empty=
+                    False,  # Here, "e" is the output of "func" so it is real data and we store it
```
Comment (mgoin) [inline vllm/model_executor/layers/parameters/lazy_compressed.py:91]: this is pretty strange formatting if you could find a different way to comment

File: vllm/model_executor/layers/parameters/lazy_compressed.py (hunk 1 above)
Comment (alexm-nm) [reply vllm/model_executor/layers/parameters/lazy_compressed.py:74]: Fixed

Comment (alexm-nm) [reply vllm/model_executor/layers/parameters/lazy_compressed.py:74]: empty(..) works, empty_like does not work here (I think empty_like is relaying on .data field that does not exist here)

**Comment by alexm-nm** [comment]
Addressed comments

File: vllm/model_executor/layers/parameters/lazy_compressed.py [hunk 3]
```diff
@@ -30,12 +31,20 @@ def __new__(cls,
...
         self.compress_transposed = compress_transposed
-        self._is_param = True
+        self.compressed_data = None
+
+        if is_empty:
+            self.uncompressed_data = None
+            self.is_empty = True
+        else:
+            self.uncompressed_data = uncompressed_data
+            self.is_empty = False
```
[the remark this answers is not in the document]
Comment (alexm-nm) [reply vllm/model_executor/layers/parameters/lazy_compressed.py:47]: Added

**Review by mgoin — APPROVED** [review]
thanks!

**Pull request merged**